db/migrate/20211201154845_add_unique_indexes.rb

```diff
@@ -0,0 +1,114 @@
+class AddUniqueIndexes < ActiveRecord::Migration[6.0]
+  include ActionView::Helpers::TextHelper
+  def up
+    clean_unindexed_tables
+    add_index :katello_capsule_lifecycle_environments, [:capsule_id, :lifecycle_environment_id], unique: true, name: 'katello_capsule_lifecycle_environments_unique_index'
+    add_index :katello_content_view_erratum_filter_rules, [:errata_id, :content_view_filter_id], unique: true, name: 'katello_content_view_erratum_filter_rules_unique_index'
+    add_index :katello_content_view_module_stream_filter_rules, [:module_stream_id, :content_view_filter_id], unique: true, name: 'katello_content_view_module_stream_filter_rules_unique_index'
+    add_index :katello_content_view_package_group_filter_rules, [:uuid, :content_view_filter_id], unique: true, name: 'katello_content_view_package_group_filter_rules_unique_index'
+    add_index :katello_content_view_repositories, [:content_view_id, :repository_id], unique: true, name: 'katello_content_view_repositories_unique_index'
+    add_index :katello_content_views, [:name, :organization_id], unique: true, name: 'katello_content_views_name_unique_index'
+  end
+
+  def down
+    remove_index :katello_capsule_lifecycle_environments, name: 'katello_capsule_lifecycle_environments_unique_index'
+    remove_index :katello_content_view_erratum_filter_rules, name: 'katello_content_view_erratum_filter_rules_unique_index'
+    remove_index :katello_content_view_module_stream_filter_rules, name: 'katello_content_view_module_stream_filter_rules_unique_index'
+    remove_index :katello_content_view_package_group_filter_rules, name: 'katello_content_view_package_group_filter_rules_unique_index'
+    remove_index :katello_content_view_repositories, name: 'katello_content_view_repositories_unique_index'
+    remove_index :katello_content_views, name: 'katello_content_views_name_unique_index'
+  end
+
+  def clean_unindexed_tables
```

Comment:
Would recommend moving cleaning and duplicating logic this to a new class along the lines of https://github.com/Katello/katello/blob/master/app/lib/katello/util/hostgroup_facets_helper.rb

Idea here is you can do something like `DeduplicationMigrator.new.execute!`

The big benefit is then you can write unit tests to test individual behavior for that specific class.

Reply:
@parthaa updated!